include/oneapi/dpl/experimental/kt/esimd_radix_sort_cooperative.h

```diff
@@ -22,19 +22,25 @@
 namespace oneapi::dpl::experimental::esimd::impl
 {
 
+constexpr uint32_t SYNC_BUFFER_PER_STAGE = 4;
+
+template <uint32_t STAGES>
 void inline init_global_sync(uint32_t * psync, uint32_t tg_id) {
     using namespace __ESIMD_NS;
     using namespace __ESIMD_ENS;
-    simd<uint32_t, 16> lane_id(0, 4);
-    simd<uint32_t, 16> old_value = lsc_atomic_update<atomic_op::load, uint32_t, 16>(psync, lane_id, 1);
+
+    constexpr uint32_t SYNC_BUFFER_SIZE = SYNC_BUFFER_PER_STAGE * STAGES;
+    
+    simd<uint32_t, SYNC_BUFFER_SIZE> lane_id(0, 4);
```

Comment:
Is 4 here the same constant?

Reply:
I think - no: as I understand ```0``` - value of first item of simd, ```4``` - step to increment each next value.

Reply:
But is not the step supposed to be equal to SYNC_BUFFER_PER_STAGE?

Reply:
You are right. Fixed.

Reply:
No, I analyzed it and I think I was wrong :) and this `4` really means `sizeof(uint32_t)`.